### PR TITLE
refactor(background): Move async initialization to background.ts to avoid nullable objects in main classes

### PR DESCRIPTION
**Pull request by melink14**
- Passes in async objects to constructors after waiting for them in background.ts
- Move config update logic to configuration.ts now that RcxDict doesn't reference RcxMain.

### CODE REVIEW

File: extension/options.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import 'lit-toast/lit-toast.js';
-import { Config, getCurrentConfiguration } from './configuration';
+import { Config, configPromise } from './configuration';
 import { LitElement, TemplateResult, css, html } from 'lit';
 import { until } from 'lit/directives/until.js';
 
+type Mutable<T> = {
+  -readonly [P in keyof T]: T[P];
+};
```
Comment (Stephie) [inline extension/options.ts:8]: Nope.

File: extension/data.ts [hunk 2]
```diff
@@ -97,12 +97,16 @@ class RcxDict {
   radData: string[] = [];
   difReasons: string[] = [];
   difRules: DeinflectionRuleGroup[] = [];
+  config: Config;
 
-  private constructor() {}
+  private constructor(initialConfig: Config) {
```
Comment (Stephie) [inline extension/data.ts:102]: `initialConfig` > `config`

File: extension/configuration.ts [hunk 3]
```diff
@@ -29,42 +29,49 @@ const DefaultConfig = {
     { code: 'U', name: 'Unicode', shouldDisplay: true },
   ],
 };
-type Config = typeof DefaultConfig;
+type InternalConfig = typeof defaultConfig;
+type Config = Readonly<InternalConfig>;
```
[the remark this answers is not in the document]
Comment (melink14) [reply extension/configuration.ts:33]: Went with MutableConfig

**Review by Stephie — APPROVED** [review]
Okay!

**Pull request merged**